src/formatDate.js

```diff
@@ -7,8 +7,62 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparatorFormat = fromFormat[3];
+  const toSeparatorFormat = toFormat[3];
+  const dateObject = {};
+
+  const arrayFromDate = date.split(fromSeparatorFormat);
+
+  fromFormat.forEach((part, i) => {
+    switch (part) {
+      case 'YYYY':
+        dateObject.year = arrayFromDate[i];
+        break;
+
+      case 'YY':
+        const year = parseInt(arrayFromDate[i]);
+
+        dateObject.year =
+          year < 30 ? '20' + arrayFromDate[i] : '19' + arrayFromDate[i];
+        break;
+
+      case 'MM':
+        dateObject.month = arrayFromDate[i];
+        break;
+
+      case 'DD':
+        dateObject.day = arrayFromDate[i];
+        break;
+
+      default:
+        break;
+    }
+  });
```

Comment:
Instead of using a `forEach` loop with a `switch` statement inside it, you could directly map the `fromFormat` parts to their corresponding values in `dateObject`. This would streamline the code and make it easier to understand.